File: e2e/java/src/test/java/dev/kreuzberg/e2e/CookiesTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: cookies. */
class CookiesTest {
    @Test
    void testCookiesPerDomain() throws Exception {
        // Isolates cookies per domain during crawl
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'cookies.length' not available on result type
        // skipped: field 'cookies' not available on result type
    }

    @Test
    void testCookiesPersistence() throws Exception {
        // Maintains cookies across multiple crawl requests
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'cookies' not available on result type
    }

    @Test
    void testCookiesSetCookieResponse() throws Exception {
        // Respects Set-Cookie header from server responses
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'cookies' not available on result type
    }

}

File: e2e/go/cookies_test.go
// E2e tests for category: cookies
package e2e_test

import (
	"strings"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_CookiesPerDomain(t *testing.T) {
	// Isolates cookies per domain during crawl
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'cookies.length' not available on result type
	// skipped: field 'cookies' not available on result type
}

func Test_CookiesPersistence(t *testing.T) {
	// Maintains cookies across multiple crawl requests
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'cookies' not available on result type
}

func Test_CookiesSetCookieResponse(t *testing.T) {
	// Respects Set-Cookie header from server responses
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'cookies' not available on result type
}
